db/migrate/20190722082648_add_derived_estimated_hours_to_work_packages.rb

```diff
@@ -0,0 +1,112 @@
+class AddDerivedEstimatedHoursToWorkPackages < ActiveRecord::Migration[5.2]
+  class WorkPackageWithRelations < ActiveRecord::Base
+    self.table_name = WorkPackage.table_name
+
+    scope :with_children, ->(*args) do
+      rel = Relation.table_name
+      wp = WorkPackage.table_name
+
+      query = "EXISTS (SELECT 1 FROM #{rel} WHERE #{rel}.from_id = #{wp}.id AND #{rel}.hierarchy > 0 LIMIT 1)"
+
+      where(query)
+    end
+  end
```

Comment:
Good craftsmanship of creating an AR class for the migration. But if you go this far to isolate the migration, I think that it would have actually been better to not rely on the `table_name` methods as those might change independent of this migration. Using the strings directly would be safe.